core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -436,7 +436,7 @@ class AdminClientIntegrationTest extends KafkaServerTestHarness with Logging {
     } catch {
       case e: ExecutionException =>
         assertTrue(e.getCause.isInstanceOf[InvalidTopicException])
```

Comment:
I'm still not clear on the difference between `InvalidTopicException` and `UnknownTopicOrPartitionException`. Yesterday you asked me to convert `InvalidTopicException` → `UnknownTopicOrPartitionException`. Can you explain?

Reply:
Sure.

`UnknownTopicOrPartition` is used if the topic doesn't exist. Because of delayed propagation of metadata, it's a retriable exception. That means that clients will generally retry if they see that exception (hopefully until a timeout, but there are cases where no timeout is applied).

The AdminClient doesn't do that yet, but I filed a JIRA for it: https://issues.apache.org/jira/browse/KAFKA-5950

`InvalidTopicException` is thrown if the topic name is invalid (say you try to create a topic with invalid characters).

We should be using `UnknownTopicOrPartitionException`, but I'm a bit uncertain because of the retriable behaviour is probably not desired in this case (once KAFKA-5950 is implemented). We should make the retriable aspect dynamic for this (and others) exception(s) to fix this sort of issue, but it hasn't been done yet. I didn't do it in this PR, but we should probably change it from `InvalidTopicException` to `UnknownTopicOrPartitionException` for consistency with other protocol APIs.